ui/src/components/EditReference/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import * as Interfaces from '@interfaces';
+import * as Components from '@components';
+import * as tiptap from '@tiptap/react';
+import * as Config from '@config';
+
+import StarterKit from '@tiptap/starter-kit';
+import Underline from '@tiptap/extension-underline';
+interface MenuBarProps {
+    editor: tiptap.Editor;
+    loading: boolean;
+    setLoading: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+type Props = {
+    positiveActionCallback: (referenceID: string, referenceContent: string) => void;
+    negativeActionCallback: () => void;
+    MenuBar: React.FC<MenuBarProps>;
+    reference: Interfaces.Reference;
+    loading: boolean;
+};
+
+const EditReference: React.FC<Props> = (props): React.ReactElement => {
```

Comment:
If this component has it's own folder (which is a good thing) we also need to extract AddReferences component into it's own folder as this is a copy of that.